avocado/core/plugin_interfaces.py

```diff
@@ -185,6 +185,12 @@ class ResultEvents(JobPreTests, JobPostTests):
     """
 
     @abc.abstractmethod
+    def start_tests(self, result):
+        """
+        Event triggered when the tests are about to start
+        """
```

Comment:
OK it looks like we don't have a method where `result` would be already initialized and tests were not yet executed and this in-deed requires it. @clebergnu you should probably also check this out, whether you had something else in mind with your job API rfc which would cover this...